clang/lib/Driver/Driver.cpp

```diff
@@ -7877,6 +7880,11 @@ Action *Driver::BuildOffloadingActions(Compilation &C,
         break;
       }
 
+      // Backend/Assemble actions are not used for the SYCL device side
```

Comment:
Change (7883-7887) provided by @mdtoguchi. Thanks Mike. For SYCL offloading, we want -c compilation to NOT invoke backend tools. For new offloading model, we see that the backend tools are being called for -c compilation. This change  prevents the backend tools from being invoked for -c compilation.